broker.go

```diff
@@ -180,12 +176,7 @@ func (b *Broker) Close() error {
 	b.done = nil
 	b.responses = nil
 
-	if b.id >= 0 {
-		b.conf.MetricRegistry.Unregister(getMetricNameForBroker("incoming-byte-rate", b))
-		b.conf.MetricRegistry.Unregister(getMetricNameForBroker("request-rate", b))
-		b.conf.MetricRegistry.Unregister(getMetricNameForBroker("outgoing-byte-rate", b))
-		b.conf.MetricRegistry.Unregister(getMetricNameForBroker("response-rate", b))
-	}
+	b.unregisterMetrics()
```

Comment:
I am not sure you need to maintain the list of registered metrics here, could you replace this method with:

```suggestion
	b.conf.MetricRegistry.UnregisterAll()
```

The MetricRegistry instance of scoped to a single broker instance, so once `broker.Close()` is called, we can Unregister all metrics attached to that broker.

Reply:
Wouldn't `UnregisterAll()` also unregister the global metrics shared between brokers?